src/overlays/actors/ovl_En_Hs2/z_en_hs2.c

```diff
@@ -27,25 +29,132 @@ const ActorInit En_Hs2_InitVars = {
     (ActorFunc)EnHs2_Update,
     (ActorFunc)EnHs2_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Hs2/EnHs2_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Hs2/EnHs2_Destroy.s")
+static ColliderCylinderInit sCylinderInit = {
+    { COLTYPE_UNK10, 0x00, 0x11, 0x39, 0x10, COLSHAPE_CYLINDER },
+    { 0x00, { 0x00000000, 0x00, 0x00 }, { 0xFFCFFFFF, 0x00, 0x00 }, 0x00, 0x01, 0x01 },
+    { 40, 40, 0, { 0, 0, 0 } },
+};
+
+static Vec3f D_80A6F4CC = { 300.0f, 1000.0f, 0.0f };
+
+extern SkeletonHeader D_06006260;
+extern AnimationHeader D_060005C0;
+
+void EnHs2_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnHs2* this = THIS;
+    SkelAnime* skelAnime = &this->skelAnime;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, &ActorShadow_DrawFunc_Circle, 36.0f);
+    SkelAnime_InitSV(globalCtx, skelAnime, &D_06006260, &D_060005C0, this->limbDrawTable,
+                     this->transitionDrawTable, 16);
+    SkelAnime_ChangeAnimDefaultRepeat(skelAnime, &D_060005C0);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    this->actor.colChkInfo.mass = 255;
```

Comment:
mass 0xFE and 0xFF are special cases so we keep them as hex. any other mass tho feel free to change to decimal